tests/test_admin/test_view.py

```diff
@@ -1418,6 +1443,90 @@ def test_post_invalid_email_error_skips_all(self):
         line_error_message = ValidationMessages.INVALID_EMAIL.format(argument=invalid_email)
         self._assert_line_message(bulk_upload_errors[0], 3, line_error_message)
 
+    @mock.patch("enterprise.utils.create_manual_enrollment_audit")
+    @mock.patch("enterprise.admin.views.create_manual_enrollment_audit")
+    @mock.patch("enterprise.models.CourseCatalogApiClient")
+    @mock.patch("enterprise.api_client.lms.EnrollmentApiClient")
+    @mock.patch("enterprise.models.EnterpriseCatalogApiClient")
+    @mock.patch("enterprise.admin.views.enroll_users_in_course")
+    @ddt.data(
+        (False, False, 200),
+        (False, True, 200),
+        (True, False, 200),
+        (True, True, 302),  # all input is valid, expect a redirect
+    )
+    @ddt.unpack
+    def test_post_create_course_enrollments(
+        self,
+        valid_course_id,
+        course_in_catalog,
+        expected_status_code,
+        enroll_users_in_course_mock,
+        enterprise_catalog_client,
+        enrollment_client,
+        course_catalog_client,
+        create_manual_enrollment_audit_mock,
+        utils_manual_enrollment_audit_mock,
+    ):
+        course_id = "course-v1:edX+DemoX+Demo_Course"
+        should_expect_enrollment_creation = False
+        self._login()
+        user = UserFactory()
+
+        enrollment_instance = enrollment_client.return_value
+        course_catalog_instance = course_catalog_client.return_value
+        enterprise_catalog_instance = enterprise_catalog_client.return_value
+        enterprise_catalog_instance.enterprise_contains_content_items.return_value = course_in_catalog
+        create_manual_enrollment_audit_mock.return_value = True
+        enroll_users_in_course_mock.return_value = [user], [], []
+
+        if valid_course_id:
+            enrollment_instance.get_course_details.side_effect = fake_enrollment_api.get_course_details
+            course_catalog_instance.get_course_run.return_value = {
+                'key': course_id,
+                'title': 'Fake Course',
+                'start': '2020-11-01T00:00:00Z',
+            }
+        else:
+            enrollment_instance.get_course_details.return_value = {}
+
+        columns = [ManageLearnersForm.CsvColumns.EMAIL, ManageLearnersForm.CsvColumns.COURSE_ID]
+        data = [
+            (user.email, course_id),
+        ]
```

Comment:
It would be good to have a test that runs this on more than one row and more than one course id.